lib/ansible/galaxy/collection/__init__.py

```diff
@@ -606,6 +606,7 @@ def verify_collections(
                 # NOTE: Verify local collection exists before
                 # NOTE: downloading its source artifact from
                 # NOTE: a galaxy server.
+                artifact = True
```

Comment:
Instead of having a boolean, how about having a default `err_msg` defined here? Then, you'd redefine it in the loop and just use whatever in the var in the `else:` block. Would this be simpler?